File: Exercícios/numeroprimo.c
#include <stdio.h>
#include <locale.h>

int main()
{
    setlocale(LC_ALL, ("pt-br"));

    int checar_numero, contador = 0;

    printf("Insira um número inteiro: ");
    scanf("%d", &checar_numero);

    for (int i = 1; i <= checar_numero; i++)
    {
        if (checar_numero % i == 0)
        {
            contador++;
        }
    }

    for (int i = 1; i <= checar_numero; i++)
    {
        if (contador > 2)
        {
            printf("O número %d, não é primo!", checar_numero);
            break;
        }
        else
        {
            printf("O número %d, é primo!", checar_numero);
            break;
        }
    }

    return 0;
}

File: Exercícios/numeroprimo.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Insira um número inteiro: ");
        int checarNumero = scanner.nextInt();
        int contador = 0;

        for (int i = 1; i <= checarNumero; i++) {
            if (checarNumero % i == 0) {
                contador++;
            }
        }

        if (contador > 2) {
            System.out.println("O número " + checarNumero + " não é primo!");
        } else {
            System.out.println("O número " + checarNumero + " é primo!");
        }

        scanner.close();
    }
}